server/src/main/java/org/elasticsearch/index/mapper/DocumentParser.java

```diff
@@ -106,11 +115,61 @@ private static void internalParseDocument(RootObjectMapper root, MetadataFieldMa
             parseObjectOrNested(context, root);
         }
 
+        executeIndexTimeScripts(context);
+
         for (MetadataFieldMapper metadataMapper : metadataFieldsMappers) {
             metadataMapper.postParse(context);
         }
     }
 
+    private static void executeIndexTimeScripts(ParseContext context) {
+        Map<String, IndexTimeScript> indexTimeScripts = context.mappingLookup().indexTimeScripts();
+        if (indexTimeScripts.isEmpty()) {
+            return;
+        }
+        SearchLookup searchLookup = new SearchLookup(
+            context.mappingLookup().indexTimeLookup()::get,
+            (ft, s) -> ft.fielddataBuilder(context.indexSettings().getIndex().getName(), s).build(
```

Comment:
I find myself wondering what `s` is, as well as `k` and `c` or `pc` in other places. Can we use slightly longer variable names so it's easier to quickly know what these variable hold?

Reply:
Yeah this is a bad habit of mine, I'll make it clearer